src/stories/Forms/Select/Select.tsx

```diff
@@ -0,0 +1,489 @@
+
+  useOnClickOutside(selectWrapperRef, () => setIsOpen(false));
+
+  const handleValueSelect = (event) => {
+    const { target } = event;
+    const selectedValue = target.dataset.value;
+    if (!selectedValue) {
+      console.error('Select component: value is not defined', target);
+      return;
+    }
+    setFilterSearch('');
+    setIsOpen(false);
+    setSelected(selectedValue);
+    selectButtonRef?.current?.focus();
+    onChange &&
+      onChange({ target: { name: selectName, id, value: selectedValue } } as React.ChangeEvent<HTMLInputElement>);
+  };
+
+  const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+    if (e.currentTarget.contains(e.relatedTarget)) return;
+    onBlur && onBlur({ target: { name: selectName, id, value: selected } } as React.FocusEvent<HTMLInputElement>);
+  };
+
+  const selectContent = getLabelOfSelectedValue(children, selected);
+  const selectContentAsString = getSelectContentAsString(selectContent);
+  const filteredChildren = getChildrenByQuery(children, filterSearch);
+  const filteredChildrenEmpty = React.Children.toArray(filteredChildren).length === 0;
+
+  return (
+    <Wrapper className={className} ref={selectWrapperRef} onBlur={handleBlur}>
+      <SelectButton
+        aria-activedescendant={selected}
+        aria-errormessage={ariaErrorMessage}
+        aria-expanded={isOpen}
+        aria-haspopup='listbox'
+        aria-labelledby={`${ariaDescribedBy || ''} ${id}-select-button`}
+        aria-owns={`${id}-select-list`}
+        hasError={hasError}
+        hasLabel={!!selectContent}
+        id={`${id}-select-button`}
+        isOpen={isOpen}
+        onClick={() => setIsOpen(!isOpen)}
+        ref={selectButtonRef}
+        title={selectContentAsString}
+        type='button'
+        value={selected}
+      >
+        {isOpen ? (
+          <FilterSearch
+            aria-autocomplete='list'
+            aria-controls={`${id}-select-list`}
+            onChange={(e) => setFilterSearch(e.target.value)}
+            placeholder='Search options ...'
+            ref={filterSearchRef}
+            type='text'
+            tabIndex={-1}
+            value={filterSearch}
+          />
+        ) : (
+          <SelectedContent hasNestedContent={typeof selectContent !== 'string'}>
+            {selectContent || placeholder}
+          </SelectedContent>
+        )}
+        <Icon name='ChevronDown' className='chevron' aria-hidden='true' />
+      </SelectButton>
+
+      {children && (
+        <SelectListWrapper
+          aria-activedescendant={selected}
+          aria-errormessage={ariaErrorMessage}
+          aria-expanded={isOpen}
+          aria-labelledby={ariaDescribedBy}
+          aria-multiselectable='false'
+          height={selectListHeight}
+          id={`${id}-select-list`}
+          isOpen={isOpen}
+          role='listbox'
+          tabIndex={isOpen ? 0 : -1}
+        >
+          <SelectListHoverBackground ref={selectListHoverBackgroundRef} aria-hidden />
```

Comment:
Note to self, explain me the need of a specific component to do hover background

Reply:
Added a comment to the component.